### ISSUE TITLE
PHP 8.0 support

### ISSUE THREAD

**Issue by turbo124** [AuthorizeNet/sdk-php#407]
I see there were issues with PHP 7.4 here #383, when running my tests against PHP 8.0 i also see the following:

```
PHPUnit\Framework\Exception: PHP Fatal error:  Cannot use "parent" when current class scope has no parent in vendor/authorizenet/authorizenet/lib/net/authorize/api/contract/v1/MerchantAuthenticationType.php on line 29
```

**Comment by mav2287** [comment]
Looks like this issue has been around since Jan 1 of 2020 with not official fix yet. Have look at: PHP 7.4 support #383 & Cannot use "parent" when current class scope has no parent #397

**Comment by turbo124** [comment]
@mav2287 I was able to use #397 as a workaround, however this is no longer possible with PHP 8.0

**Comment by mav2287** [comment]
Have you taken a look at the php-version-upgrade branch and this pull request? Updating generator to fix PHP74 related bug #395

**Comment by mav2287** [comment]
Just tried switching to that branch with `composer require authorizenet/authorizenet:dev-php-version-upgrade`, deleted the vendor folder and then ran `composer install` , but I still get the same parent error. It would seem that didn't fix it.

**Comment by turbo124** [comment]
@mav2287 I believe it is the stricter type checks in PHP 8

**Comment by mav2287** [comment]
That would make sense. I dropped a comment on the merge request for the person who made the 7.4 patch to see if they are doing an 8.0 patch. If they do hopefully it gets merged in as well.

**Comment by skystebnicki** [comment]
Have there been any updates on this? We've been holding off on upgrading to php8 because of this bug, but it is starting to create some problems for us.

**Comment by turbo124** [comment]
@skystebnicki have you tested @markonikolicdir PR as a fix for this? I am also in a similar position as this is the only package that is holding us back from PHP 8

**Comment by skystebnicki** [comment]
Unfortunately we were unable to get it to work, even with the PR fix, but we didn't spend too much time testing that because it's pretty critical we have the official package tested and working in our production environment.

**Comment by brada1703** [comment]
Any update on this? Running into the same problem on 8.0. I'd love to use the [fix here](https://github.com/vkhramtsov/sdk-php/tree/bugfix-remove-parent-call-when-class-does-not-have-parent), but can't until it's official :(

**Comment by midnightcorp** [comment]
+1 for PHP 8 support.

**Comment by kkeith-adg** [comment]
Been trying to upgrade our app to PHP 8 for the last few days and this library is one of my main issues. Decided to write a PHP script to patch the files so I wouldn't have to use a fork of the official library. If you are looking for a temporary solution until they either fix this issue or you migrate away from Authorize.net (recommended), this could work.

1. Setup [this script](https://gist.github.com/kkeith-adg/f29cb3008c2888d330a40688b4c65d50) in the same directory as your `composer.json` file
2. Run via `php name-of-script.php`
3. Install as a composer `post-install-cmd` script in your `composer.json`
```
    "scripts": {
        "post-install-cmd": [
            "php name-of-script.php"
        ]
    }
```

**Comment by skystebnicki** [comment]
Is anyone from Authorize.net even monitoring this repo? Maybe we're just screaming in the desert here. Seems like such a trivial fix to leave undone for so long.

**Comment by skystebnicki** [comment]
I just filed a ticket with their support to see if we can get some eyes on it. Note of businesses out there, if you publish a github repo, it's not a good look if you fail to respond to customers there.

**Comment by papercheck** [comment]
I posted the issue in the Authorize.net Developer Community forum under [PHP 8 Support for AuthorizeNet/sdk-php](https://community.developer.authorize.net/t5/Integration-and-Testing/PHP-8-Support-for-AuthorizeNet-sdk-php/m-p/75477/highlight/true#M47066).

**Comment by kkeith-adg** [comment]
Looks like this was fixed in version 2.0.2

**Comment by turbo124** [comment]
@kkeith-adg thanks for updating. All my tests are looking green now.
